src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -160,6 +161,29 @@ public AzureAppConfigurationOptions Select(string keyFilter, string labelFilter
             return this;
         }
 
+        /// <summary>
+        /// Specify a snapshot and include its contained key-values in the configuration provider.
+        /// <see cref="SelectSnapshot"/> can be called multiple times to include key-values from multiple snapshots.
+        /// </summary>
+        /// <param name="name">The name of the snapshot in Azure App Configuration.</param>
+        public AzureAppConfigurationOptions SelectSnapshot(string name)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (!_kvSelectors.Any(s => !string.IsNullOrEmpty(s.SnapshotName) && s.SnapshotName.Equals(name)))
```

Comment:
`!string.IsNullOrEmpty(s.SnapshotName)`

That check doesn't seem necessary.

Reply:
SnapshotName can be null if Select is called before SelectSnapshot. Wasn't sure of a different way to handle it

Reply:
if (!_kvSelectors.Any(s => string.Equals(s.SnapshotName, name)))

Reply:
Thanks!